### PR TITLE
Add methods for setting, getting and uploading avatar

### PR DESCRIPTION
**Pull request by agraven**

### CODE REVIEW

File: matrix_sdk/src/client.rs [hunk 1]
```diff
@@ -478,6 +478,66 @@ impl Client {
...
+    /// # block_on(async {
+    /// # let homeserver = Url::parse("http://locahost:8080").unwrap();
+    /// # let client = Client::new(homeserver).unwrap();
+    /// let path = Path::from("/home/example/selfie.jpg");
+    /// let mut image = File::open(path).unwrap();
+    ///
+    /// client.upload_avatar(&mime::IMAGE_JPG, &mut image).await.expect("Can't set avatar");
+    /// #})
+    /// ```
+    pub async fn upload_avatar<R: Read>(&self, content_type: &Mime, reader: &mut R) -> Result<()> {
```
Comment (poljar) [inline matrix_sdk/src/client.rs:534]: Can the avatar URL be of any mime type? Perhaps we should restrict this to image formats or perhaps restricting anything here is anyways futile. Any thoughts on this?

**Review by poljar — CHANGES_REQUESTED** [review]
The examples fail to compile for various reasons, e.g. missing closing delimiter.

The doctests get executed when you run `cargo test` as well, but I should really swap out our CI.

File: matrix_sdk/src/client.rs (hunk 1 above)
Comment (agraven) [reply matrix_sdk/src/client.rs:534]: I looked through the client-server spec and couldn't find any mentioned of requirements for mime types of avatars, so i figure it shouldn't be the sdk's role to restrict those

**Comment by agraven** [comment]
Ah, i was trying to figure out how to run those doc tests but the help info for cargo gave me nothing and looking it up didn't help me either.

**Comment by agraven** [comment]
Doc tests should be fixed now

**Pull request merged**